eclipse-collections/src/main/java/org/eclipse/collections/impl/list/immutable/ImmutableArrayList.java

```diff
@@ -282,13 +282,29 @@ public T detect(Predicate<? super T> predicate)
     }
 
     @Override
+    public <P> T detectWith(Predicate2<? super T, ? super P> predicate, P parameter)
+    {
+        return InternalArrayIterate.detectWith(this.items, this.items.length, predicate, parameter);
+    }
+
+    @Override
     public T detectIfNone(Predicate<? super T> predicate, Function0<? extends T> function)
     {
         T result = this.detect(predicate);
         return result == null ? function.value() : result;
     }
 
     @Override
+    public <P> T detectWithIfNone(
+            Predicate2<? super T, ? super P> predicate,
+            P parameter,
+            Function0<? extends T> function)
+    {
+        T result = this.detectWith(predicate, parameter);
+        return result == null ? function.value() : result;
+    }
```

Comment:
The override for detectWith() looks fine because it delegates to InternalArrayIterate. I don't understand why detectWithIfNone() is improved by an override. And detectIfNone() is identical to the method it overrides, should we remove it?